Makefile

```diff
@@ -1,53 +1,105 @@
+pip_install_c = $(pip_install) -c requirements.txt -c 'https://docs.inmanta.com/inmanta-service-orchestrator-dev/$(ISO_VERSION)/reference/requirements.txt'
+else
+pip_install_c = $(pip_install) -c requirements.txt
+endif
+
+src_dirs := src tests $(shell test -d tests_common && echo tests_common || true)
 
-.PHONY: install
+
+.PHONY: install ci-install
 install:
-	pip install -U setuptools pip uv -c requirements.txt
-	uv pip install -U -e . -c requirements.txt -r requirements.dev.txt
+	$(pip_install_c) -U setuptools pip uv
+	$(pip_install_c) -U -e .[dev]
+
+ci-install:
+ifeq ($(shell which uv),)
+	$(error uv is required for this target.)
+endif
+ifndef ISO_VERSION
+	$(error ISO_VERSION make variable needs to be set for this target. Run `make ISO_VERSION=<x> $@`.)
+endif
+	$(pip_install_c) -U setuptools pip uv
+# some tests are skipped for editable installs => no editable on ci
+	$(pip_install_c) -U .[dev]
 
 .PHONY: install-tests
 install-tests:
-	pip install -U setuptools pip
-	python3 tests_common/copy_files_from_core.py
-	pip install -e ./tests_common
+	$(pip_install) -U setuptools pip uv
+	$(PYTHON) tests_common/copy_files_from_core.py
+	$(pip_install) -e ./tests_common
 
 .PHONY: format
 format:
 	$(isort)
 	$(black)
 
-.PHONY: pep8
+.PHONY: pep8 ci-pep8
 pep8:
-	uv pip install -c requirements.txt pep8-naming flake8-black flake8-isort
-	flake8 src tests tests_common
+	$(PYTHON) -m flake8 $(src_dirs)
+ci-pep8:
+	$(PYTHON) -m flake8 --output-file flake8-report.txt --tee $(src_dirs)
+	$(PYTHON) -m junit_conversor flake8-report.txt junit-pep8.xml
 
-RUN_MYPY=MYPYPATH=stubs:src python -m mypy --soft-error-limit=-1 --html-report mypy -p inmanta
+parser:=src/inmanta/parser/
+parsetab:=$(parser)/parsetab.py
+# load inmanta.app so that the parser is generated for a consistent mypy baseline
+$(parsetab): $(parser)/plyInmantaLex.py $(parser)/plyInmantaParser.py
+	$(PYTHON) -m inmanta.app >/dev/null
+	touch $@
 
-.PHONY: mypy
-mypy:
-	$(RUN_MYPY) | mypy-baseline filter
+.PHONY: mypy ci-mypy
+mypy: $(parsetab)
+	$(mypy) | $(mypy_baseline) filter --sort-baseline
+ci-mypy: $(parsetab)
+	$(mypy) --junit-xml junit-mypy.xml --cobertura-xml-report coverage | $(mypy_baseline) filter --no-colors --sort-baseline
 
 .PHONY: mypy-sync
-mypy-sync:
-	$(RUN_MYPY) | mypy-baseline sync
+mypy-sync: $(parsetab)
+	$(mypy) | $(mypy_baseline) sync --sort-baseline
 
 .PHONY: mypy-full
-mypy-full:
-	$(RUN_MYPY)
+mypy-full: $(parsetab)
+	$(mypy)
 
-.PHONY: test
+.PHONY: test ci-test
 test:
-	pytest -vvv --log-level DEBUG tests
+	$(PYTHON) -m pytest -vvv --log-level DEBUG $(TESTS)
+ci-test:
+	$(PYTHON) -m pytest -vvv --log-level DEBUG -p no:sugar --junitxml=junit-tests.xml $(PYTEST_EXTRA_ARGS) $(TESTS)
+
+.PHONY: all ci-all
+all: pep8 mypy test
+ci-all: ci-pep8 ci-mypy ci-test
 
-.PHONY: testcov
-testcov:
-	pytest --cov=inmanta --cov-report html:coverage --cov-report term -vvv tests
+venv-%: FORCE $(shell mktemp -d)/bin/python install %
+	rm -rf $(<:/bin/python=)
```

Comment:
bug: `$<` now resolves to `FORCE`, not `<tmpdir>/bin/python` as intended. So we try to `rm -rf FORCE` and never clean up the venv. Will fix.